internal/diffgenerator/diffgenerator.go

```diff
@@ -0,0 +1,197 @@
+// FIXME: Give me a better package name
+package diffgenerator
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"io"
+	"math/rand/v2"
+)
+
+type RecordType uint8
+
+const (
+	RecordTypeUpdated RecordType = iota
+	RecordTypeZero
+)
+
+type DataRecord struct {
+	Type   RecordType
+	Offset uint64
+	Length uint64
+	Data   []byte
```

Comment:
We should prepare a constructor function to create a `DataRecord` instance so that users don't have to remember which field should be populated.